sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/resolver/FunctionResolver.scala

```diff
@@ -77,9 +77,14 @@ class FunctionResolver(
 
   private val typeCoercionResolver: TypeCoercionResolver =
     new TypeCoercionResolver(timezoneAwareExpressionResolver)
+  private val expressionResolutionContextStack =
+    expressionResolver.getExpressionResolutionContextStack
 
   /**
    * Main method used to resolve an [[UnresolvedFunction]]. It resolves it in the following steps:
+   *  - Check if the `unresolvedFunction` is an aggregate expression. Set
```

Comment:
```suggestion
   *  - Check if the `unresolvedFunction` is an [[AggregateExpression]]. Set
```
Otherwise it's easy to confuse `AggregateExpression` with one of the `Aggregate.aggregateExpressions`.